packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/dynamoDb-walkthrough.ts

```diff
@@ -61,490 +93,430 @@ async function updateWalkthrough(context, defaultValuesFilename, serviceMetadata
-  );
-  print.info('');
-  print.info('To learn more about primary keys, see:');
-  print.info(
-    'https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.CoreComponents.html#HowItWorks.CoreComponents.PrimaryKey',
-  );
-  print.info('');
-  // Ask for primary key
-
-  answers.KeySchema = [];
-  let partitionKeyName;
-  let partitionKeyType;
-
-  if (resourceName) {
-    ({ partitionKeyName } = defaultValues);
-    ({ partitionKeyType } = defaultValues);
-  } else {
-    const primaryKeyQuestion = {
-      type: inputs[4].type,
-      name: inputs[4].key,
-      message: inputs[4].question,
-      validate: amplify.inputValidation(inputs[3]),
-      choices: indexableAttributeList,
-    };
-
-    const partitionKeyAnswer = await inquirer.prompt([primaryKeyQuestion]);
-
-    partitionKeyName = partitionKeyAnswer[inputs[4].key];
+  if (!cliInputs.resourceName) {
+    throw new Error('resourceName not found in cli-inputs');
   }
 
-  answers.KeySchema.push({
-    AttributeName: partitionKeyName,
-    KeyType: 'HASH',
-  });
+  const { attributeAnswers, indexableAttributeList } = await askAttributeListQuestion(context, existingAttributeDefinitions);
 
-  // Get the type for primary index
+  cliInputs.gsi = await askGSIQuestion(context, indexableAttributeList, attributeAnswers, cliInputs.gsi);
+  cliInputs.triggerFunctions = await askTriggersQuestion(context, cliInputs.resourceName, cliInputs.triggerFunctions);
 
-  const primaryAttrTypeIndex = answers.AttributeDefinitions.findIndex(attr => attr.AttributeName === partitionKeyName);
+  cliInputsState.saveCliInputPayload(cliInputs);
 
-  partitionKeyType = answers.AttributeDefinitions[primaryAttrTypeIndex].AttributeType;
+  const stackGenerator = new DDBStackTransform(cliInputs.resourceName);
+  stackGenerator.transform();
 
-  usedAttributeDefinitions.add(partitionKeyName);
+  return cliInputs;
+}
 
-  let sortKeyName;
-  let sortKeyType;
+async function askTriggersQuestion(context: $TSContext, resourceName: string, existingTriggerFunctions?: string[]): Promise<string[]> {
+  const { amplify } = context;
+  let triggerFunctions: string[] = existingTriggerFunctions || [];
 
-  if (resourceName) {
-    ({ sortKeyName } = defaultValues);
+  if (!existingTriggerFunctions || existingTriggerFunctions.length === 0) {
+    if (await amplify.confirmPrompt('Do you want to add a Lambda Trigger for your Table?', false)) {
+      let triggerName;
+      try {
+        // @ts-expect-error ts-migrate(2554) FIXME: Expected 3 arguments, but got 2.
+        triggerName = await addTrigger(context, resourceName);
+        return [triggerName];
+      } catch (e) {
+        printer.error(e.message);
+      }
+    }
+  } else {
+    const triggerOperationQuestion = {
+      type: 'list',
+      name: 'triggerOperation',
+      message: 'Select from the following options',
+      choices: ['Add a Trigger', 'Remove a trigger', 'Skip Question'],
```

Comment:
Suggest changing the last option to "I'm done"

Reply:
Changed. Good suggestion.